# Perkenalan

Nama  : Vicky Eldora Wuisan

Program ini dibuat menganalysis data penjualan seblak dari ratingnya, harganya, banyak terjualnya, kota penjualnya. Untuk nantinya saya dapat berjualan seblak dengan mengetahui bagaimana marketnya di tokopedia agar saat berjualan nantinya saya sudah mengetahui harus dijual dengan harga berapa agar dapat laku dan menguntungkan. Data-data tersebut diambil dengan Web Scraping website tokopedia kategori seblak.

# Library

In [2]:
pip install bs4 selenium

Note: you may need to restart the kernel to use updated packages.


In [4]:
pip install requests

Note: you may need to restart the kernel to use updated packages.


In [3]:
# import library
import pandas as pd
from bs4 import BeautifulSoup
from selenium import webdriver
import time
from scipy import stats
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# A. Web Scraping

In [141]:
url="https://www.tokopedia.com/search?navsource=&page=1&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st="
driver = webdriver.Chrome()
driver.get(url)

html = driver.page_source
soup = BeautifulSoup(html, "html.parser")

print(soup.prettify()[:700])

In [142]:
soup.find_all('div', {"class":"prd_link-product-name css-3um8ox"})

In [143]:
Nama_Produk = []
Harga_Produk = []
Penjual = []
Kota_Toko = []
Banyaknya_Terjual = []
Rating_Produk = []

for i in range(1, 11):
    url = "https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st=".format(i)
    # Ekstrak halaman   
    driver.get(url)
    # kasih waktu buat looping
    time.sleep(0.1)
    # Ekstrak HTML
    html = driver.page_source
    soup = BeautifulSoup(html, "html.parser")

    # Loop melalui setiap barang pada halaman
    for seblak in soup.find_all('div', {"class": "css-1asz3by"}):
        try:
            Nama_Produk.append(seblak.find('div', {"class":"prd_link-product-name css-3um8ox"}).get_text())
        except AttributeError:
            Nama_Produk.append(None)
        try:
            Harga_Produk.append(seblak.find('div', {"class":"prd_link-product-price css-h66vau"}).get_text())
        except AttributeError:
            Harga_Produk.append(None)
        try:
            Penjual.append(seblak.find('span',{"class":"prd_link-shop-name css-1kdc32b flip"}).get_text())
        except AttributeError:
            Penjual.append(None)
        try:
            Kota_Toko.append(seblak.find('span',{"class":"prd_link-shop-loc css-1kdc32b flip"}).get_text())
        except AttributeError:
            Kota_Toko.append(None)
        try:
            Banyaknya_Terjual.append(seblak.find('span',{"class":"prd_label-integrity css-1sgek4h"}).get_text())
        except AttributeError:
            Banyaknya_Terjual.append(None)
        try:
            Rating_Produk.append(seblak.find('span',{"class":"prd_rating-average-text css-t70v7i"}).get_text())
        except AttributeError:
            Rating_Produk.append(None)


In [144]:
# Buat dataframe kosong
df = pd.DataFrame()

# Isi dataframe hasil webscraping
df['Nama_Produk'] = Nama_Produk
df['Harga_Produk'] = Harga_Produk
df['Penjual'] = Penjual
df['Kota_Toko'] = Kota_Toko
df['Banyaknya_Terjual'] = Banyaknya_Terjual
df['Rating_Produk'] = Rating_Produk

df

# B. Data Preparation

In [146]:
df.to_csv('P0G3_vicky_eldora',index=False)

In [5]:
# menampilkan data
df = pd.read_csv('P0G3_vicky_eldora.csv')
df

Nama_Produk Harga_Produk  \
0                              Kylafood Seblak karuhun     Rp11.880   
1                   Kylafood Seblak Mie Baso Sapi Play     Rp24.200   
2    Kylafood Mix Fav Selebgram Seblak Original & B...     Rp44.340   
3                             Kylafood Seblak Original     Rp15.750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
..                                                 ...          ...   
362                      [250 gr] Kerupuk Seblak Pedas     Rp11.500   
363  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...     Rp15.000   
364  Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
365           Seblak Instan Komplit Pedas Gurih Nikmat      Rp5.999   
366                      Seblak Instan Pedas Home Made      Rp3.500   

            Penjual          Kota_Toko Banyaknya_Terjual  Rating_Produk  
0               NaN                NaN     4 rb+ terjual            4.8  
1               NaN                NaN      250+ terjual            4.9  
2               NaN                NaN      100+ terjual            4.8  
3          kylafood            Bandung     10rb+ terjual            4.9  
4          Lidigeli         Kab. Garut      100+ terjual            4.8  
..              ...                ...               ...            ...  
362     CamilanJati      Kab. Sidoarjo       50+ terjual            5.0  
363        Lidigeli         Kab. Garut      250+ terjual            4.8  
364        Lidigeli         Kab. Garut      100+ terjual            4.8  
365  Baso Aci Ayang         Kab. Garut      3rb+ terjual            4.9  
366      the Dhecip  Tangerang Selatan      3rb+ terjual            4.9  

[367 rows x 6 columns]

## 1. Eksplorasi data sederhana

In [6]:
# menampilkan beberapa baris data
df.head(7)

Nama_Produk Harga_Produk  \
0                            Kylafood Seblak karuhun     Rp11.880   
1                 Kylafood Seblak Mie Baso Sapi Play     Rp24.200   
2  Kylafood Mix Fav Selebgram Seblak Original & B...     Rp44.340   
3                           Kylafood Seblak Original     Rp15.750   
4  Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
5             Seblak Rafael/Seblak Coet Instan Halal     Rp26.600   
6  Seblak Instan Kenyang Murah Asli Bandung Resel...      Rp7.138   

               Penjual        Kota_Toko Banyaknya_Terjual  Rating_Produk  
0                  NaN              NaN     4 rb+ terjual            4.8  
1                  NaN              NaN      250+ terjual            4.9  
2                  NaN              NaN      100+ terjual            4.8  
3             kylafood          Bandung     10rb+ terjual            4.9  
4             Lidigeli       Kab. Garut      100+ terjual            4.8  
5           Foodstocks  Jakarta Selatan       60+ terjual            5.0  
6  Berkah Food Bandung           Cimahi      7rb+ terjual            4.9

Dari beberapa baris data terdapat banyak nama produk dari seblak seperti seblak rafael, seblak ceker, seblak mie baso. Harga produk yang beragam juga dari paling murah Rp 7.138 dan termahal Rp44.340. Tidak semua produk terdapat nama penjual dan kota toko, Produk yang terjual ada yang mencapai ribuan, dari data saya lihat harga produk yang mahal terjualnya tidak sebanyak harga yang murah dan rating dari produk paling kecil 4.8.

In [7]:
# menampilkan ringkasan singkat tentang data saya, rangeindex, total kolom, nama kolom, berupa jumlah nilai bukan nol di setiap kolom dan tipe data kolom
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Nama_Produk        367 non-null    object 
 1   Harga_Produk       367 non-null    object 
 2   Penjual            350 non-null    object 
 3   Kota_Toko          350 non-null    object 
 4   Banyaknya_Terjual  367 non-null    object 
 5   Rating_Produk      357 non-null    float64
dtypes: float64(1), object(5)
memory usage: 17.3+ KB


In [8]:
# menampilkan jenis types data
df.dtypes

Nama_Produk           object
Harga_Produk          object
Penjual               object
Kota_Toko             object
Banyaknya_Terjual     object
Rating_Produk        float64
dtype: object

Data yang saya dapat setelah webscraping sebanyak 367 data. Dari tipe data ada yang masih belum sesuai seperti harga produk dan banyaknya terjual harusnya berupa tipe data integer atau float, langkah selanjutnya adalah saya akan mengubah tipe data tersebut terlebih dahulu. Kemudian melakukan cleaning data pada langkah selanjutnya

In [9]:
# cek missing value
df.isnull().sum()

Nama_Produk           0
Harga_Produk          0
Penjual              17
Kota_Toko            17
Banyaknya_Terjual     0
Rating_Produk        10
dtype: int64

karena masih terdapat missing value pada kolom penjual, kota toko, dan rating produk, langkah selanjutnya adalah mengdrop missing value tersebut dengan df.dropna()

## 2. Data Cleaning

In [10]:
# hapus missing value
df = df.dropna()

In [11]:
#cek missing value lagi ---> sudah tidak ada lagi missing value
df.isnull().sum()

Nama_Produk          0
Harga_Produk         0
Penjual              0
Kota_Toko            0
Banyaknya_Terjual    0
Rating_Produk        0
dtype: int64

In [12]:
# menghapus data duplicates
df = df.drop_duplicates()

In [13]:
# data yang telah dihapus missing value dan duplicatesnya
df

Nama_Produk Harga_Produk  \
3                             Kylafood Seblak Original     Rp15.750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
5               Seblak Rafael/Seblak Coet Instan Halal     Rp26.600   
6    Seblak Instan Kenyang Murah Asli Bandung Resel...      Rp7.138   
7                                  SEBLAK INSTAN MOMMY     Rp13.900   
..                                                 ...          ...   
306                   Seblak Koin Warna Odjolali Snack     Rp11.900   
337                              SEBLAK INSTAN KOMPLIT      Rp6.500   
339  KriCa (Keripik Kaca)/ Seblak Beling / Keripik ...     Rp12.000   
340                         Kerupuk Seblak Pedas 200gr     Rp12.250   
341      Kylafood Paket 2 Seblak Cup + 2 Basreng pedas     Rp47.260   

                 Penjual        Kota_Toko Banyaknya_Terjual  Rating_Produk  
3               kylafood          Bandung     10rb+ terjual            4.9  
4               Lidigeli       Kab. Garut      100+ terjual            4.8  
5             Foodstocks  Jakarta Selatan       60+ terjual            5.0  
6    Berkah Food Bandung           Cimahi      7rb+ terjual            4.9  
7       Rav Jaya Display   Kab. Tangerang      1rb+ terjual            4.8  
..                   ...              ...               ...            ...  
306       Odjolali Snack          Bandung      100+ terjual            5.0  
337      Vandiva Kuliner      Kab. Malang         8 terjual            5.0  
339       Garut Beunghar       Kab. Garut       40+ terjual            5.0  
340          Utama.Snack         Surabaya      750+ terjual            4.9  
341             kylafood          Bandung         3 terjual            5.0  

[91 rows x 6 columns]

In [14]:
# melakukan replace data 
df['Harga_Produk'] = df['Harga_Produk'].str.replace('Rp','').str.replace('.','')
df['Banyaknya_Terjual'] = df['Banyaknya_Terjual'].str.replace('rb','000').str.replace(' ','').str.replace('+','').str.replace('terjual','')

In [15]:
# merubah data type dari object menjadi float
df['Harga_Produk'] = df['Harga_Produk'].astype(int)
df['Banyaknya_Terjual'] = df['Banyaknya_Terjual'].astype(int)

In [16]:
# menampilkan informasi data dari data yang telah di cleaning
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 91 entries, 3 to 341
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Nama_Produk        91 non-null     object 
 1   Harga_Produk       91 non-null     int32  
 2   Penjual            91 non-null     object 
 3   Kota_Toko          91 non-null     object 
 4   Banyaknya_Terjual  91 non-null     int32  
 5   Rating_Produk      91 non-null     float64
dtypes: float64(1), int32(2), object(3)
memory usage: 4.3+ KB


tipe data sudah sesuai setelah dilakukan data cleaning

In [17]:
# menampilkan data yang telah di cleaning
df

Nama_Produk  Harga_Produk  \
3                             Kylafood Seblak Original         15750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
5               Seblak Rafael/Seblak Coet Instan Halal         26600   
6    Seblak Instan Kenyang Murah Asli Bandung Resel...          7138   
7                                  SEBLAK INSTAN MOMMY         13900   
..                                                 ...           ...   
306                   Seblak Koin Warna Odjolali Snack         11900   
337                              SEBLAK INSTAN KOMPLIT          6500   
339  KriCa (Keripik Kaca)/ Seblak Beling / Keripik ...         12000   
340                         Kerupuk Seblak Pedas 200gr         12250   
341      Kylafood Paket 2 Seblak Cup + 2 Basreng pedas         47260   

                 Penjual        Kota_Toko  Banyaknya_Terjual  Rating_Produk  
3               kylafood          Bandung              10000            4.9  
4               Lidigeli       Kab. Garut                100            4.8  
5             Foodstocks  Jakarta Selatan                 60            5.0  
6    Berkah Food Bandung           Cimahi               7000            4.9  
7       Rav Jaya Display   Kab. Tangerang               1000            4.8  
..                   ...              ...                ...            ...  
306       Odjolali Snack          Bandung                100            5.0  
337      Vandiva Kuliner      Kab. Malang                  8            5.0  
339       Garut Beunghar       Kab. Garut                 40            5.0  
340          Utama.Snack         Surabaya                750            4.9  
341             kylafood          Bandung                  3            5.0  

[91 rows x 6 columns]

# C. Business Understanding/Problem Statement

**Problem Statment**

Meningkatkan penjualan produk seblak di Tokopedia menjadi lebih tinggi penjualannya.

**`SMART`**

> Specific
* Meningkatkan penjualan produk seblak di Tokopedia.

> Measurment
* Menargetkan peningkatan 20% kenaikan penjualan produk seblak dari jumlah produk seblak yang terjual sebelumnya.

> Achievable
* Melakukan peningkatan kualitas produk seblak selama 1.5 bulan pertama, kemudian melakukan analisis data yang tepat untuk memahami preferensi atau keinginan konsumen di 1.5 bulan berikutnya, dan di bulan ke 4 melakukan optimasi strategi pemasaran online (paid advertisement di social media atau endosrment).

> Relevant
* Peningkatan kualitas seblak, dan optimasi strategi pemasaran akan memberikan stimulasi terhadap target pasar, dan memberikan dampak positif untuk produk seblak.

> Time-Bound
* Mencapai peningkatan penjualan minimal 20% dalam waktu empat bulan.

> Conclusions
* meningkatkan penjualan seblak di tokopedia dengan melakukan improvement produk, analisis keingin konsumen, dan promosi sehingga penjualan dapat naik 20% dalam waktu 4 bulan.

# D.  Analysis

## 1. Analysis diawali dengan menghitung nilai mean, median, modus, standar deviasi, kurtosis dari kolom harga, banyak produk terjual, dan rating produk untuk mengetahui bentuk data apakah ada anomali atau tidak.

### Harga_Produk

In [18]:
# menghitung mean,median, dan modus
Harga_Produk_mean = df['Harga_Produk'].mean()
Harga_Produk_median = df['Harga_Produk'].median()
Harga_Produk_modus = df['Harga_Produk'].mode().values[0]

# menghitung standar deviasi
std_Harga_Produk = df['Harga_Produk'].std()
variance_Harga_Produk = df['Harga_Produk'].var()

# skewness
Skewness_Harga_Produk =  df['Harga_Produk'].skew()

# kurtosis
Kurtosis_Harga_Produk =  df['Harga_Produk'].kurtosis()

print('Mean:',Harga_Produk_mean)
print('Median:',Harga_Produk_median)
print('Modus:',Harga_Produk_modus)
print('Standard Deviation:',std_Harga_Produk)
print('Skewness:',Skewness_Harga_Produk)
print('Kurtosis:' , Kurtosis_Harga_Produk)

Mean: 17863.362637362636
Median: 13999.0
Modus: 9000
Standard Deviation: 12727.594332801715
Skewness: 1.4991102929778277
Kurtosis: 2.2235415582717484


dari kolom harga produk rata-rata harga seblak adalah `17863`, nilai tengah harga seblak `13999`, harga seblak banyak dijual di harga `9000`, standar deviasi sebesar `12727` yang artinya terdapat banyak variasi harga produk , skewness sebesar `1.49911` lebih dari 1 artinya positive skew atay highly skewed, dan kurtosis sebesar `2.22354` artinya memiliki bentuk leptokurtic dimana memiliki heavy tails dan higher peak.

### Banyaknya_Terjual

In [19]:
# menghitung mean,median, dan modus
Banyaknya_Terjual_mean = df['Banyaknya_Terjual'].mean()
Banyaknya_Terjual_median = df['Banyaknya_Terjual'].median()
Banyaknya_Terjual_modus = df['Banyaknya_Terjual'].mode().values[0]

# menghitung standar deviasi
std_Banyaknya_Terjual = df['Banyaknya_Terjual'].std()

# skewness
Skewness_Banyaknya_Terjual =  df['Banyaknya_Terjual'].skew()

# kurtosis
Kurtosis_Banyaknya_Terjual =  df['Banyaknya_Terjual'].kurtosis()

print('Mean:',Banyaknya_Terjual_mean)
print('Median:',Banyaknya_Terjual_median)
print('Modus:',Banyaknya_Terjual_modus)
print('Standard Deviation:',std_Banyaknya_Terjual)
print('Skewness:',Skewness_Banyaknya_Terjual)
print('Kurtosis:' , Kurtosis_Banyaknya_Terjual)

Mean: 721.2087912087912
Median: 80.0
Modus: 100
Standard Deviation: 1867.2983664254593
Skewness: 3.7657727846211646
Kurtosis: 14.65039290461223


dari kolom banyaknya terjual rata-rata banyaknya terjual adalah `721.2`, nilai tengah banyaknya terjual `80`, kebanyakan toko telah menjual `100` barang, standar deviasi sebesar `1867.2` yang artinya terdapat banyak variasi dari banyaknya terjual setiap toko, skewness sebesar `3.7657` lebih dari 1 artinya positive skew atay highly skewed, dan kurtosis sebesar `14.650` artinya memiliki bentuk leptokurtic dimana memiliki heavy tails dan higher peak.

### Rating_Produk

In [20]:
# menghitung mean,median, dan modus
Rating_Produk_mean = df['Rating_Produk'].mean()
Rating_Produk_median = df['Rating_Produk'].median()
Rating_Produk_modus = df['Rating_Produk'].mode().values[0]

# menghitung standar deviasi
std_Rating_Produk = df['Rating_Produk'].std()

# skewness
Skewness_Rating_Produk =  df['Rating_Produk'].skew()

# kurtosis
Kurtosis_Rating_Produk =  df['Rating_Produk'].kurtosis()

print('Mean:',Rating_Produk_mean)
print('Median:',Rating_Produk_median)
print('Modus:',Rating_Produk_modus)
print('Standard Deviation:',std_Rating_Produk)
print('Skewness:',Skewness_Rating_Produk)
print('Kurtosis:' , Kurtosis_Rating_Produk)

Mean: 4.890109890109889
Median: 4.9
Modus: 5.0
Standard Deviation: 0.14380151834691005
Skewness: -2.0695280800060716
Kurtosis: 5.057122413329102


dari kolom rating produk rata-rata rating produk adalah `4.890`, nilai tengah rating produk `4.9`, kebanyakan toko mendapat rating `5`, standar deviasi sebesar `0.14` yang artinya variasi dari rating produk setiap toko hampir mirip (tidak berbeda jauh), skewness sebesar `-2.06` lebih dari -1 artinya negative skew atay highly skewed, dan kurtosis sebesar `5.057` artinya memiliki bentuk leptokurtic dimana memiliki heavy tails dan higher peak.

In [21]:
# menampilkan describe dari data (count,mean,std,min,max,25%,50%,75%)
df.describe()

Harga_Produk  Banyaknya_Terjual  Rating_Produk
count     91.000000          91.000000      91.000000
mean   17863.362637         721.208791       4.890110
std    12727.594333        1867.298366       0.143802
min     2500.000000           1.000000       4.300000
25%     9850.000000          30.000000       4.800000
50%    13999.000000          80.000000       4.900000
75%    24112.500000         375.000000       5.000000
max    64000.000000       10000.000000       5.000000

## 2. kemudian analysis berikutnya adalah ingin mengetahui jika saya mecoba menjual seblak dengan permisalan confidence 95% dan permisalan data terdistribusi normal, kira-kira berapa potensi minimum dan maksimum pendapatan dari mejual seblak tersebut.

In [22]:
# membuat kolom baru pendapatan dengan banyaknya terjual dikali dengan harga produk
df['Pendapatan'] = df['Banyaknya_Terjual']*df['Harga_Produk']

# menghitung std kemudian menghitung lower dan upper limit
std = df['Pendapatan'].std()
N = len(df)
low, up = stats.norm.interval(0.95,loc=df['Pendapatan'].mean(),scale=std/np.sqrt(N))
print('Lower Limit:',low)
print('Upper Limit:',up)

Lower Limit: 2921511.179371205
Upper Limit: 10853461.084365059


potensi minimum pendapatan `2.921.511` dan maksimum pendapatan `10.853.461` jika produk terjual merupakan penjualan produk perbulan dengan confidence level 95%

## 3. Menganalysis harga seblak antara di jabodetabek dan di luar jabodetabek berbeda atau sama. Dengan pemikirain bahwa pasti biaya bahan baku di tiap daerah berbeda.

In [23]:
# menampilkan value dari kolom Kota_Toko
df['Kota_Toko'].unique()

array(['Bandung', 'Kab. Garut', 'Jakarta Selatan', 'Cimahi',
       'Kab. Tangerang', 'Depok', 'Jakarta Timur', 'Tangerang Selatan',
       'Surabaya', 'Jakarta Barat', 'Kab. Bogor', 'Surakarta',
       'Kab. Bandung', 'Kab. Sidoarjo', 'Bekasi', 'Semarang', 'Tangerang',
       'Kab. Majalengka', 'Jakarta Pusat', 'Tasikmalaya', 'Kab. Jombang',
       'Banjarbaru', 'Serang', 'Kab. Malang'], dtype=object)

analisis harga antara daerah jabodetabek dan daerah luar jabodetabek

In [24]:
# menampilkan tabel jumlah value dari banyaknya produk terjual dengan hanya menampilkan kolom kota_toko dan banyaknya_terjual dengan diurutkan berdasarkan nama toko
kota_terjual = df[['Kota_Toko','Banyaknya_Terjual']].groupby('Kota_Toko').sum()
kota_terjual

Banyaknya_Terjual
Kota_Toko                           
Bandung                        10274
Banjarbaru                        11
Bekasi                            28
Cimahi                         14000
Depok                            740
Jakarta Barat                   5796
Jakarta Pusat                      6
Jakarta Selatan                 1690
Jakarta Timur                  13280
Kab. Bandung                     100
Kab. Bogor                       215
Kab. Garut                      6371
Kab. Jombang                      11
Kab. Majalengka                    8
Kab. Malang                        8
Kab. Sidoarjo                     50
Kab. Tangerang                  1004
Semarang                         250
Serang                             8
Surabaya                        1490
Surakarta                        100
Tangerang                        406
Tangerang Selatan               9754
Tasikmalaya                       30

saya akan menggunakan `Two-Sample Independent Test`. karena ada dua kondisi

In [25]:
# menampilkan tabel dengan hanya kota jabodetabek
jabodetabek = df[(df["Kota_Toko"]=='Jakarta Selatan') | 
                 (df["Kota_Toko"]=='Kab. Tangerang') | 
                 (df["Kota_Toko"]=='Depok') | 
                 (df["Kota_Toko"]=='Jakarta Timur') | 
                 (df["Kota_Toko"]=='Tangerang Selatan') | 
                 (df["Kota_Toko"]=='Jakarta Barat') | 
                 (df["Kota_Toko"]=='Kab. Bogor') |
                 (df["Kota_Toko"]=='Bekasi') |
                 (df["Kota_Toko"]=='Tangerang') |
                 (df["Kota_Toko"]=='Jakarta Pusat')]

jabodetabek.head()

Nama_Produk  Harga_Produk  \
5    Seblak Rafael/Seblak Coet Instan Halal         26600   
7                       SEBLAK INSTAN MOMMY         13900   
8   Seblak Rafael, Seblak Coet Instan Halal         23750   
9                   SEBLAK INSTANT TERMURAH          4450   
10            Seblak Instan Pedas Home Made          3500   

                           Penjual          Kota_Toko  Banyaknya_Terjual  \
5                       Foodstocks    Jakarta Selatan                 60   
7                 Rav Jaya Display     Kab. Tangerang               1000   
8                Brother Meat Shop              Depok                500   
9   Pusat Cuankie & Cemilan Frozen      Jakarta Timur              10000   
10                      the Dhecip  Tangerang Selatan               3000   

    Rating_Produk  Pendapatan  
5             5.0     1596000  
7             4.8    13900000  
8             4.9    11875000  
9             4.8    44500000  
10            4.9    10500000

In [26]:
# menampilkan tabel dengan hanya kota luar jabodetabek
non_jabodetabek = df[(df["Kota_Toko"]=='Bandung') | 
                       (df["Kota_Toko"]=='Kab. Garut') | 
                       (df["Kota_Toko"]=='Cimahi') | 
                       (df["Kota_Toko"]=='Surabaya') | 
                       (df["Kota_Toko"]=='Surakarta') | 
                       (df["Kota_Toko"]=='Kab. Sidoarjo') | 
                       (df["Kota_Toko"]=='Semarang') |
                       (df["Kota_Toko"]=='Kab. Majalengka') |
                       (df["Kota_Toko"]=='Tasikmalaya') |
                       (df["Kota_Toko"]=='Kab. Jombang') |
                       (df["Kota_Toko"]=='Banjarbaru') |
                       (df["Kota_Toko"]=='Serang') |
                       (df["Kota_Toko"]=='Kab. Malang') |
                       (df["Kota_Toko"]=='Kab. Bandung')]

non_jabodetabek.head()

Nama_Produk  Harga_Produk  \
3                            Kylafood Seblak Original         15750   
4   Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
6   Seblak Instan Kenyang Murah Asli Bandung Resel...          7138   
11   Cuanki Lidah Topping Baso Aci Seblak Isi 50 Biji         13000   
21  SEBRING KRUPUK KERUPUK SEBLAK KERING PEDAS DAU...         16000   

                Penjual   Kota_Toko  Banyaknya_Terjual  Rating_Produk  \
3              kylafood     Bandung              10000            4.9   
4              Lidigeli  Kab. Garut                100            4.8   
6   Berkah Food Bandung      Cimahi               7000            4.9   
11          Pedas Gilaa    Surabaya                 60            4.9   
21          Aydaa Snack   Surakarta                100            4.9   

    Pendapatan  
3    157500000  
4      2400000  
6     49966000  
11      780000  
21     1600000

In [27]:
# nama kota yang ada di jabodetabek
jabodetabek_nama = ['Jakarta Selatan', 'Kab. Tangerang', 'Depok', 'Jakarta Timur', 
'Tangerang Selatan', 'Jakarta Barat', 'Kab. Bogor',
'Bekasi', 'Tangerang', 'Jakarta Pusat']

# menghitung rata-rata harga produk dari kota jabodetabek
jabodetabek = df[df["Kota_Toko"].isin(jabodetabek_nama)]
jabodetabek_rata2 = jabodetabek['Harga_Produk'].mean()
print(f'rata-rata harga produk di jabodetabek =', jabodetabek_rata2)

rata-rata harga produk di jabodetabek = 18638.88888888889


In [28]:
# nama kota yang ada di luar jabodetabek
non_jabodetabek_nama = ['Bandung', 'Kab. Garut', 'Cimahi',
'Surabaya',  'Surakarta',
'Kab. Sidoarjo',  'Semarang', 
'Kab. Majalengka',  'Tasikmalaya', 'Kab. Jombang',
'Banjarbaru', 'Serang', 'Kab. Malang', 'Kab. Bandung']

# menghitung rata-rata harga produk dari kota luar jabodetabek
non_jabodetabek = df[df["Kota_Toko"].isin(non_jabodetabek_nama)]
non_jabodetabek_rata2 = non_jabodetabek['Harga_Produk'].mean()
print(f'rata-rata harga produk di jabodetabek =', non_jabodetabek_rata2)

rata-rata harga produk di jabodetabek = 16731.513513513513


rata-rata harga seblak di jabodetabek = `18638.88` dan rata-rata harga seblak di luar jabodetabek = `16731.51`, tidak berbeda jauh hanya beda 2000. tapi agar lebih yakin langkah selanjutnya dibuktikan dengan menghitung Two-Sample Independent Test

hipotesis :

> H0: Harga produk di jabodetabek sama dengan harga produk di luar jabodetabek

> H1: Harga produk di jabodetabek tidak sama dengan harga produk di luar jabodetabek

In [29]:
t_stat, p_val = stats.ttest_ind(jabodetabek['Harga_Produk'],non_jabodetabek['Harga_Produk'])
print('T-Statistic:',t_stat)
print('P-value:',p_val)

T-Statistic: 0.7002185869489953
P-value: 0.4856172192252417


p-value lebih besar dari 0.05 saya menyimpulkan bahwa tidak dapat menolak hipotenis null, yang berarti tidak terdapat perbedaan harga yang terlalu jauh atau harga bisa saja sama antara harga produk di jabodetabek dan harga produk di luar jabodetabek.

## 4. Kemudian muncul pertanyaan berikutnya setelah mengetahui harga antara jabodetabek dan luar jabodetabek hampir sama, apakah produk dengan harga yang lebih murah akan lebih banyak terjual?

In [30]:
# memakai sprearmanr karena kedua data distribusi tidak normal
corr_rho, pval_s = stats.spearmanr(df['Harga_Produk'], df['Banyaknya_Terjual'])

print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")

rho-correlation: -0.45, p-value: 6.275982461404158e-06


p-value digunakan untuk memberikan informasi apakah korelasi tersebut nyata atau hanya kebetulan atau tidak signifikan. Jika p-value  > 0.05 maka hanya kebetulan atau tidak signifikan, sedangkan jika p-value < 0.05 maka korelasi nyata.

p-value <  0.05 maka korelasi antara harga produk dan banyaknya terjual bersifat `nyata`. Tetapi korelasi negattif yang artinya semakin murah harga produk seblak maka semakin banyak yang terjual.

# E. Conclusion

Berdasarkan analisis yang telah dilakukan penjualan seblak di tokopedia:
 * rata-rata harga seblak yang dijual di tokopedia adalah Rp17.863
 * dengan rata-rata ratingnya 4.89, dapat dikatakan orang-orang menyukai produk seblak yang dijual
 * dengan rata-rata seblak terjual adalah 721.2 seblak, dapat dikatakan banyak orang yang membeli seblak di tokopedia
 * lebih banyak seblak yang terjual jika harganya lebih murah
 * Penjualan di jabodetabek dan di luar jabodetabek memiliki harga yang sama ataupun hampir sama walaupun biaya bahan baku di kedua lokasi berbeda
 * Jika saya memulai menjual seblak di tokopedia pendapatan minimal saya adalah Rp2.921.511 dan pendapatan maksimumnya Rp10.853.461 dalam waktu 1 bulan

> Dari data-data diatas saya dapat memulai berjualan seblak dengan mengetahui bagaimana marketnya di tokopedia sehingga saat berjualan saya sudah mengetahui harus dijual dengan harga berapa, variasi harga yang saya bisa gunakan agar seblak yang saya jual nanti dapat laku dan menguntungkan.